packages/wallet-service/src/api/wallet.ts

```diff
@@ -380,6 +381,65 @@ interface LoadResult {
   xpubkey: string;
 }
 
+/*
+ * This lambda will be started by a SNSMessage on the load failed SNS configured
+ * in serverless.yml. It will receive all wallet load failed events published on
+ * the loadWalletAsync DLQ SNS topic
+ */
+export const loadWalletFailed: Handler<SNSEvent> = async (event) => {
+  const logger = createDefaultLogger();
+  const records = event.Records;
+
+  try {
+    for (let i = 0; i < records.length; i++) {
+      const snsEvent = records[i].Sns;
+      const { RequestID, ErrorMessage } = snsEvent.MessageAttributes;
+
+      // Process each failed load wallet event
+      const loadEvent: LoadEvent = JSON.parse(snsEvent.Message) as unknown as LoadEvent;
+
+      if (!loadEvent.xpubkey) {
+        logger.error('Received wallet load fail message from SNS but no xpubkey received');
+        continue;
+      }
+
+      const walletId = getWalletId(loadEvent.xpubkey);
+
+      // update wallet status to 'error' and set the number of retries to MAX so
+      // it doesn't get retried
+      await updateWalletStatus(mysql, walletId, WalletStatus.ERROR, MAX_LOAD_WALLET_RETRIES);
+
+      logger.error(`${walletId} failed to load.`);
+      logger.error({
+        xpubkey: loadEvent.xpubkey,
+        walletId,
+        RequestID: RequestID.Value,
+        ErrorMessage: ErrorMessage.Value,
+      });
+
+      await addAlert(
+        'A wallet failed to load in the wallet-service',
+        `The wallet with id ${walletId} failed to load on the wallet-service. Please check the logs.`,
+        Severity.MINOR,
+        {
+          xpubkey: loadEvent.xpubkey,
+          walletId,
+          RequestID: RequestID.Value,
+          ErrorMessage: ErrorMessage.Value,
+        },
+      );
```

Comment:
[doubt] What is the purpose of this alert, to build awareness? What are the possible causes for this issue?